Formula/webkitgtk.rb

```diff
@@ -0,0 +1,132 @@
+class Webkitgtk < Formula
+  desc "GTK interface to WebKit"
+  homepage "https://webkitgtk.org"
+  url "https://webkitgtk.org/releases/webkitgtk-2.36.3.tar.xz"
+  sha256 "732fcf8c4ec644b8ed28b46ebbd7c1ebab9d9e0afea9bdf5e5d12786afc478d1"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@3.10" => :build
+  depends_on "cairo"
+  depends_on "enchant"
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "gcc"
+  depends_on "glib"
+  depends_on "gst-plugins-base"
+  depends_on "gstreamer"
+  depends_on "gtk+3"
+  depends_on "harfbuzz"
+  depends_on "icu4c"
+  depends_on "jpeg-turbo"
+  depends_on "jpeg-xl"
+  depends_on "libavif"
+  depends_on "libgcrypt"
+  depends_on "libnotify"
+  depends_on "libpng"
+  depends_on "libsecret"
+  depends_on "libsoup"
+  depends_on "libxcomposite"
+  depends_on "libxt"
+  depends_on :linux # Use JavaScriptCore.Framework on macOS.
```

Comment:
This seems to support macOS. https://trac.webkit.org/wiki/BuildingGtk#BuildingtheGTKportonMacOS

Reply:
Oops, I missed your second comment here.

Reply:
No worries.  Later in that page they write:

> The MacOS build is not actively maintained to the same standards than the Linux one. So your mileage may vary.

and that was a pretty accurate reflection of how it went trying to build on macOS.

I suspect things might be more likely to work on macOS if we used a version of GTK build with the X backend instead of Quartz, but that's a bad idea for a number of reasons i.e. lack of proper HiDPI support.  We've done a pretty good job of building GTK software on macOS with the Quartz backend and I don't think we should deviate from that.

Having said this, one downside of this being Linux-only is that if someone wanted to submit a formula for any of the GTK apps that need WebKitGTK as opposed to just JavaScriptCore, those would be Linux-only apps too.  If we encounter that, we might want to open a upstream issue to ask if they are interested in getting the Quartz backend to work with OpenGL.Framework or better yet Metal, but these days I've found that the GTK developers tend to only focus on supporting GNOME and don't have much interest in making things work nicely on other Linux DEs, let alone macOS or Windows.

Reply:
> better yet Metal

This would be much better. I don't think anyone is interested in investing the time to get it done, sadly.

Reply:
> This would be much better. I don't think anyone is interested in investing the time to get it done, sadly.

Definitely agreed. I think for now we should move forward with bottling this on Linux so that we can at least provide WebKitGTK on one platform.